src/test/lib/optimizer/strategy/semi_join_reduction_rule_test.cpp

```diff
@@ -45,35 +45,53 @@ TEST_F(SemiJoinReductionRuleTest, CreateSimpleReduction) {
   // selectivity, a semi join reduction should be created.
 
   // clang-format off
-  const auto input_lqp = JoinNode::make(JoinMode::Inner, equals_(_a_a, _b_a),
+  const auto input_lqp =
+  JoinNode::make(JoinMode::Inner, equals_(_a_a, _b_a),
     _node_a,
     _node_b);
 
-  const auto expected_lqp = JoinNode::make(JoinMode::Inner, equals_(_a_a, _b_a),
-    JoinNode::make(JoinMode::Semi, equals_(_a_a, _b_a),
-      _node_a,
-      _node_b),
+  const auto expected_reduction =
+  JoinNode::make(JoinMode::Semi, equals_(_a_a, _b_a),
+    _node_a,
+    _node_b);
+
+  const auto expected_lqp =
+  JoinNode::make(JoinMode::Inner, equals_(_a_a, _b_a),
+    expected_reduction,
     _node_b);
   // clang-format on
+  expected_reduction->mark_as_semi_reduction(expected_lqp);
 
   auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
   EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+
+  // Check whether the added semi join was also marked as a semi reduction.
+  auto join_node = std::static_pointer_cast<JoinNode>(actual_lqp->left_input());
+  ASSERT_TRUE(join_node->is_semi_reduction());
```

Comment:
I'm aware that we do not use `ASSERT/EXPECT` consistently, but I'd argue that there is no semantic difference in the checks here. Thus, we should decide for one option.

Reply:
`ASSERT_*` aborts the test if it fails. But yeah, I changed it to `EXPECT_*`.

Reply:
That's why I'd use `ASSERT` only when it's preventing the following cases from segfaulting (e.g, `ASSERT` that variable is not `nullptr`, `EXPECT` its members to be equal value `x`). But I guess we do not have a guideline for that.

Reply:
Makes sense 👍